sdk/identity/identity/src/credentials/chainedTokenCredential.ts

```diff
@@ -80,7 +80,10 @@ export class ChainedTokenCredential implements TokenCredential {
     }
 
     if (!token && errors.length > 0) {
-      const err = new AggregateAuthenticationError(errors);
+      const err = new AggregateAuthenticationError(
+        errors,
+        "ChainedTokenCredential authentication failed."
+      );
```

Comment:
Without this, this error had an initial segment with an `undefined` string. This PR adds a test to ensure this is never silently broken in the future.